### PR TITLE
Add utility to `rr.components.Color` to generate colors from any string (and use it in the air traffic data example)

### PR DESCRIPTION
**Pull request by abey79**
### What

It's some time nice to log some color information in multiple entities to make it easier to relate them visually. This PR adds a `rr.components.Color.from_str()` utility that does exactly that: generate a nice color randomly picked up based on the provided string.

This PR also updates the air traffic data example so the barometric traces have matching colors with the map data.

### CODE REVIEW

File: rerun_py/rerun_sdk/rerun/components/color_ext.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+_GOLDEN_RATIO = (math.sqrt(5.0) - 1.0) / 2.0
+
+
+class ColorExt:
+    """Extension for [Color][rerun.components.Color]."""
+
+    @staticmethod
+    def from_string(s: str) -> Color:
+        """
+        Generate a random yet color based on a string.
```
Comment (teh-cmc) [inline rerun_py/rerun_sdk/rerun/components/color_ext.py:19]:
```suggestion
        Generate a random yet deterministic color based on a string.

File: examples/python/air_traffic_data/air_traffic_data.py [hunk 2]
```diff
@@ -310,7 +315,13 @@ def log_position_and_altitude(self, df: polars.DataFrame, icao_id: str) -> None:
             return
 
         if icao_id not in self._position_indicators:
-            rr.log(entity_path, [rr.archetypes.Points3D.indicator(), rr.archetypes.GeoPoints.indicator()], static=True)
+            color = rr.components.Color.from_string(entity_path)
+            rr.log(
+                entity_path,
+                [rr.archetypes.Points3D.indicator(), rr.archetypes.GeoPoints.indicator(), color],
```
Comment (teh-cmc) [inline examples/python/air_traffic_data/air_traffic_data.py:321]: Good lord I didnt realize how much that code needs tagged components :see_no_evil:

**Review by teh-cmc — APPROVED** [review]

File: examples/python/air_traffic_data/air_traffic_data.py (hunk 2 above)
Comment (abey79) [reply examples/python/air_traffic_data/air_traffic_data.py:321]: Good example of a component which is _intended_ to be shared across two archetypes, btw. Hopefully we don't make that use case too contrived, aka if that `Color` is  tagged with the `Points3D` archetype, it should be implicitly(?) used by the `GeoPoints` archetype as well.

Comment (teh-cmc) [reply examples/python/air_traffic_data/air_traffic_data.py:321]: Ha yeah I didn't even realize that was what's happening, interesting.

Yeah we'll have to figure out how we want to handle those. I do think heuristics to implicitly pick up the same component from unrelated tags if on the same entity makes sense.

Comment (abey79) [reply examples/python/air_traffic_data/air_traffic_data.py:321]: Vaguely related is one realisation we've had when discussing the archetype vs. visualiser relationship the other day: indicator components are additive, while tags are exclusive. With the former, you could have 2 archetypes with 1 components (e.g. hypothetical ScalarSeriesLine + ScalarSeriesPoint archetypes based on a single scalar component + 2 indicators), but with the latter you are forced to make a choice.

We don't think this constraint is going to be much of problem though. More in https://github.com/rerun-io/rerun/issues/8368

**Pull request merged**